.github/workflows/python-app.yml

```diff
@@ -34,15 +34,15 @@ jobs:
       - name: Install dependencies
         run: |
           python -m pip install --upgrade pip
-          pip install flake8 pytest rich pytest-pretty sphinx coverage
-          if [ -f requirements.txt ]; then pip install -r requirements.txt; fi
+          python -m pip install -r requirements.txt
+          python -m pip install flake8
```

Comment:
Personally I prefer pytest, I find them faster and easier to use, do you suggest moving to unit tests?

Reply:
I removed pytest since you are not using it. In fact, all the tests are run using unittest. So there is no need to install pytest just to have a nice command to run the test suite. And it is not listed in the requirements, so it is generally a good idea that the steps that run in the CI are (basically) the same in your local env. 
For example, you run the test locally with unittest, but then you run them in the CI with pytest. What if these two are not compatible? Let's say that there is a bug in pytest, then it would be hard to find the problem